src/thunderbots/software/ai/hl/stp/tactic/tactic.cpp

```diff
@@ -54,14 +56,13 @@ std::unique_ptr<Intent> Tactic::getNextIntent(
     {
         // Figure out where this intent should and should not move to, based on the
         // world state, whitelisted avoid areas, and blacklisted avoid areas
-        std::vector<AvoidArea> avoid_areas;
+        avoid_area_mask_t avoid_areas = 0;
         if (game_state_opt)
         {
-            avoid_areas = getAreasToAvoid(*game_state_opt);
+            avoid_areas |= getAvoidAreasFromGameState(*game_state_opt);
         }
-        avoid_areas.insert(avoid_areas.end(), blacklisted_avoid_areas.begin(),
-                           blacklisted_avoid_areas.end());
-
+        avoid_areas &= (~whitelisted_avoid_areas);
```

Comment:
is there anyway we can avoid doing these operations on a tactic level, and somehow abstract it away in the avoid area implementation / api?

Reply:
I abstracted it with MACROs but you didn't like that

Reply:
Are using MACROSs the only way this can be abstracted?

Reply:
I thought that was the best way to abstract it. Otherwise, could define a class and override +/- to do this math.